ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt

```diff
@@ -60,31 +60,40 @@ public open class HttpClientCall internal constructor(
     public lateinit var response: HttpResponse
         internal set
 
+    /**
+     * Tries to receive the payload of the [responseData] as a specific expected type provided in [info].
+     *
+     * @throws NoTransformationFoundException If no transformation is found for the type [info].
+     * */
+    internal suspend fun receiveFromData(info: TypeInfo, responseData: Any): Any {
+        val subject = HttpResponseContainer(info, responseData)
+        val currentClient = client ?: error("Failed to receive call($this) in different native thread.")
+
+        val result = currentClient.responsePipeline.execute(this, subject).response
+        if (!result.instanceOf(info.type)) {
+            val from = result::class
+            val to = info.type
+            throw NoTransformationFoundException(response, from, to)
+        }
+
+        return result
+    }
+
     /**
      * Tries to receive the payload of the [response] as a specific expected type provided in [info].
      * Returns [response] if [info] corresponds to [HttpResponse].
      *
      * @throws NoTransformationFoundException If no transformation is found for the type [info].
      * @throws DoubleReceiveException If already called [receive].
      */
-    public suspend fun receive(info: TypeInfo): Any {
+    public open suspend fun receive(info: TypeInfo): Any {
```

Comment:
Making `open` receive is complicate. Let's make an open `flag` instead to check the condition in the parent class and override it in the `SavedCall`